cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -351,5 +355,278 @@ ExtensionIdRegistry* default_extension_id_registry() {
   return &impl_;
 }
 
+Status FunctionMapping::AddArrowToSubstrait(std::string arrow_function_name, ArrowToSubstrait conversion_func){
+  if (arrow_to_substrait.find(arrow_function_name) != arrow_to_substrait.end()){
+    arrow_to_substrait[arrow_function_name] =  conversion_func;
+  }
+  return Status::OK();
+}
+
+Status FunctionMapping::AddSubstraitToArrow(std::string substrait_function_name, SubstraitToArrow conversion_func){
+  if (substrait_to_arrow.find(substrait_function_name) != substrait_to_arrow.end()){
+    substrait_to_arrow[substrait_function_name] =  conversion_func;
+  }
```

Comment:
Same, perhaps the else should be an invalid status.

Reply:
Returning a AlreadyExist status here, will that be better?